### PR TITLE
fix excludes filtering, add tests

### PR DESCRIPTION
**Pull request by FredKSchott**
- [X] CHANGELOG.md has been updated

This PR exposes a previously private method for testing purposes. I'm generally against doing this, but the alternative would mean to mock out the internal git clone logic. This seemed more reasonable, but I'm open to different solutions if this is a step to far for anyone.

### CODE REVIEW

**Comment by justinfagnani** [comment]
I'm perfectly fine with doing this. It's the easiest way to accomplish the goal for a class that we're very likely the only consumers of. It wasn't actually private before anyway, just annotated that way in TypeScript. Non-TypeScript callers wouldn't have seen an error.

File: src/workspace.ts [hunk 1]
```diff
@@ -165,20 +165,20 @@ export class Workspace {
    * wildcard-containing patterns as needed) to return full GitHub repo
    * information for all matched repos.
    */
-  private async _determineGitHubRepos(): Promise<GitHubRepo[]> {
+  async determineGitHubRepos(): Promise<GitHubRepo[]> {
```
Comment (justinfagnani) [inline src/workspace.ts]: I'd keep the underscore, it's still not meant to be publicized as part of the public API.

File: src/workspace.ts [hunk 2]
```diff
@@ -165,20 +165,20 @@ export class Workspace {
...
     const matchPatterns = this.options.match;
     const excludePatterns =
-        (this.options.exclude || []).map(String.prototype.toLowerCase);
+        (this.options.exclude ||
+         []).map(((excludePattern) => excludePattern.toLowerCase()));
```
Comment (justinfagnani) [inline src/workspace.ts]: I'd almost always prefer this style of callback. It prevents soooo many problems.

File: src/workspace.ts [hunk 3]
```diff
@@ -165,20 +165,20 @@ export class Workspace {
...
     const matchedReferences = allMatchedReferences.filter((ref) => {
       return !excludePatterns.includes(ref.fullName.toLowerCase());
     });
     // Fetch the full repo information for each matched reference
-    const matchedRepos =
-        await batchProcess(matchedReferences, async (ref) => {
-          return this._github.getRepoInfo(ref);
-        }, {concurrency: githubConcurrencyPreset});
+    const matchedRepos = await batchProcess(matchedReferences, async (ref) => {
+      return this._github.getRepoInfo(ref);
```
Comment (justinfagnani) [inline src/workspace.ts]: btw, you don't need the braces and return for this arrow function.

File: src/workspace.ts (hunk 2 above)
Comment (FredKSchott) [reply src/workspace.ts]: agreed, I keep re-learning that lesson

**Pull request merged**